prov/util/src/util_shm.c

```diff
@@ -68,21 +68,36 @@ static void smr_peer_addr_init(struct smr_addr *peer)
 void smr_cma_check(struct smr_region *smr, struct smr_region *peer_smr)
 {
 	struct iovec local_iov, remote_iov;
+	int remote_pid;
 	int ret;
-
-	if (peer_smr->cma_cap != SMR_CMA_CAP_NA) {
-		smr->cma_cap = peer_smr->cma_cap;
-		return;
+	uint8_t mask, shift;
+
+	if (smr == peer_smr) {
+		mask = SMR_CMA_SELF_MASK;
+		shift = SMR_CMA_SELF_SHIFT;
+	} else {
+		mask = SMR_CMA_PEER_MASK;
```

Comment:
I think this mask needs to be SMR_CMA_SELF_MASK and opposite for the one above since the mask here is used to save the opposite setting that we're not figuring out.

Reply:
Nevermind. I think it's ok. Sorry! Looks good to me. Thank you!